File: armstrong-number.js
var isArmstrong = function(N) {
  let armstrong = Array.from(N.toString()).map(Number).reduce((acc, cur) => {
    return acc + Math.pow(cur, N.toString().length);
  }, 0);
  return armstrong === N ? true : false;
};

console.log(isArmstrong(123));

File: armstrong-number.kt
import kotlin.math.pow

class Solution {
  fun isArmstrong(N: Int): Boolean {
    var num = N
    var sum = 0
    val mutableList = mutableListOf<Int>()
    while (num > 0) {
      mutableList.add(num % 10)
      num /= 10
    }
    mutableList.forEach {
      sum += it.toDouble().pow(mutableList.size).toInt()
    }
    return sum == N
  }
}